### PR TITLE
update SM storage-optimized instance types

### PR DESCRIPTION
**Pull request by micheleRP**
## Description
Adds Amazon storage-optimized instance types

Resolves https://redpandadata.atlassian.net/browse/DOC-742
Review deadline:

## Page previews
[Linux](https://deploy-preview-843--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/self-hosted/manual/production/requirements/#amazon)
[K8s](https://deploy-preview-843--redpanda-docs-preview.netlify.app/current/deploy/deployment-option/self-hosted/kubernetes/k-requirements/#amazon)

## Checks

- [ ] New feature
- [ ] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Review by hcoyote — APPROVED** [review]
ship it! thanks for catching this so quickly

File: modules/deploy/partials/requirements.adoc [hunk 1]
```diff
@@ -171,6 +171,8 @@ ifdef::env-kubernetes[EKS defaults to the ext4 file system. Use XFS instead wher
 
 - Storage optimized: Storage-optimized instances are designed for workloads that require high, sequential read and write access to very large data sets on local storage. They are optimized to deliver tens of thousands of low-latency, random IOPS to applications.
 
+** https://aws.amazon.com/ec2/instance-types/i4g/[i4g, is4gen, im4gn^]
```
Comment (JakeSCahill) [inline modules/deploy/partials/requirements.adoc]:
```suggestion
** https://aws.amazon.com/ec2/instance-types/i4g/[I4g, Is4gen, Im4gn^]
```

File: modules/deploy/partials/requirements.adoc [hunk 2]
```diff
@@ -171,6 +171,8 @@ ifdef::env-kubernetes[EKS defaults to the ext4 file system. Use XFS instead wher
...
+** https://aws.amazon.com/ec2/instance-types/i4i/[i4i^]
```
Comment (JakeSCahill) [inline modules/deploy/partials/requirements.adoc]:
```suggestion
** https://aws.amazon.com/ec2/instance-types/i4i/[I4i^]
```

**Review by JakeSCahill — APPROVED** [review]
Small suggestions on capitalisation to match how Amazon refers to these instances.

**Pull request merged**